frontend/src/components/FilterMenu.js

```diff
@@ -62,8 +79,14 @@ export function FilterItem({ filter, onRemoveFilter, onUpdateFilter }) {
     }
   });
 
-  const onUpdate = (name, value) => {
-    onUpdateFilter(id, name, value);
+  const onUpdate = (name, value, toggleAllChecked = true) => {
```

Comment:
This might be related to the parent PR, but whenever I select `specialist` for the first filter after loading the page "All Specialists" is populated for the query before I select the condition (`Contains` vs `Does not contain`)

Reply:
This issue is related to the parent PR, will comment there.

Reply:
As we discussed this is more of change with the base filter component.